contracts/registry/src/lib.rs

```diff
@@ -204,6 +204,125 @@ impl Contract {
         }
     }
 
+    // If it is the first iteration of the transfer, bans the source account, otherwise returns the last transfered token
+    fn get_transfer_with_continuation_info(
+        &mut self,
+        from: &AccountId,
+        to: &AccountId,
+    ) -> (bool, IssuerTokenId) {
+        match self.ongoing_soul_tx.get(from) {
+            // starting the process
+            None => (false, self.start_transfer_with_continuation(from, to)),
+            // resuming sbt_recover process
+            Some(s) => (true, s),
+        }
+    }
+
+    // execution of the sbt_recover in this function to parametrize `limit` in
+    // order to facilitate tests.
+    fn _sbt_recover(&mut self, from: AccountId, to: AccountId, limit: usize) -> (u32, bool) {
+        let storage_start = env::storage_usage();
+        let issuer = env::predecessor_account_id();
+        let issuer_id = self.assert_issuer(&issuer);
+        self.assert_not_banned(&to);
+        // if first execution ban the source account, othwerwise get the last transfered token
+        let (resumed, start) = self.get_transfer_with_continuation_info(&from, &to);
+
+        let mut tokens_recovered = 0;
+        let mut class_ids = Vec::new();
+
+        let mut last_token_transfered = BalanceKey {
+            owner: from.clone(),
+            issuer_id,
+            class_id: 0,
+        };
+
+        for (key, token) in self
+            .balances
+            .iter_from(balance_key(from.clone(), start.issuer_id, start.token))
+            .take(limit)
+        {
+            if key.owner != from || key.issuer_id != issuer_id {
+                continue;
```

Comment:
shouldn't we `break` here?

Reply:
`continue` jumps into the next loop iteration. If we break we might run into a case where we do not transfer some tokens. I noticed it while testing different scenarios